clients/src/test/java/org/apache/kafka/common/security/oauthbearer/internals/secured/RefreshingHttpsJwksTest.java

```diff
@@ -62,14 +62,14 @@ public class RefreshingHttpsJwksTest extends OAuthBearerTest {
     @Test
     public void testBasicScheduleRefresh() throws Exception {
         String keyId = "abc123";
-        Time time = new MockTime();
+        MockTime time = new MockTime();
         HttpsJwks httpsJwks = spyHttpsJwks();
 
-        try (RefreshingHttpsJwks refreshingHttpsJwks = getRefreshingHttpsJwks(time, httpsJwks)) {
+        try (RefreshingHttpsJwks refreshingHttpsJwks = getRefreshingHttpsJwks(time, httpsJwks, mockExecutorService(time))) {
             refreshingHttpsJwks.init();
             verify(httpsJwks, times(1)).refresh();
             assertTrue(refreshingHttpsJwks.maybeExpediteRefresh(keyId));
-            verify(httpsJwks, times(1)).refresh();
+            verify(httpsJwks, times(2)).refresh();
```

Comment:
This assertion was incorrect, mockito expects us to pass through the total number of times the invocation has been called.

We expect to call refresh once in `init()` and again when calling `maybeExpediteRefresh`. Since `maybeExpediteRefresh` was scheduling the refresh in a background thread, the test thread would usually win the race. We can deterministically induce a failure, by calling `Thread.sleep(100)` in the test thread.

Reply:
Thanks @splett2 - so the assertion would fail whenever maybeExpediteRefresh actually ran ! LGTM

Reply:
Oof what an unfortunate bug. Thanks for fixing.